warehouse/query-core/src/test/java/datawave/query/jexl/visitors/RangeCoalescingVisitorTest.java

```diff
@@ -113,11 +85,48 @@ public void testCoalesceManyNestedAndUnorderedTermsWithAnchorTerm() throws Parse
         String expectedQuery1 = "TACO == 'tacocat' && NUM3 == '+aE3' && (NUM2 >= '+aE2' && NUM2 <= '+aE4') && (NUM >= '+aE1' && NUM <= '+aE5')";
         String expectedQuery2 = "TACO == 'tacocat' && NUM3 == '+aE3' && (NUM >= '+aE1' && NUM <= '+aE5') && (NUM2 >= '+aE2' && NUM2 <= '+aE4')";
         
-        ASTJexlScript script = JexlASTHelper.parseJexlQuery(originalQuery);
-        ASTJexlScript newScript = RangeCoalescingVisitor.coalesceRanges(script);
+        assertVisitorResult(originalQuery, expectedQuery1, expectedQuery2);
+    }
+    
+    private void assertVisitorResult(String original, String expected) throws ParseException {
+        ASTJexlScript originalScript = JexlASTHelper.parseJexlQuery(original);
+        ASTJexlScript expectedScript = JexlASTHelper.parseJexlQuery(expected);
+        ASTJexlScript actualScript = RangeCoalescingVisitor.coalesceRanges(originalScript);
         
-        String newQuery = JexlStringBuildingVisitor.buildQuery(newScript);
-        assertTrue("Expected [" + expectedQuery1 + "] or [" + expectedQuery2 + "] but was " + newQuery,
-                        (expectedQuery1.equals(newQuery) || expectedQuery2.equalsIgnoreCase(newQuery)));
+        assertScriptEquality(actualScript, expectedScript);
+        assertTrue(JexlASTHelper.validateLineage(actualScript, true));
```

Comment:
This is 100% not required for this PR, but I would LOVE to see us validate the lineage of the originalScript after running the visitor as well.  In my opinion, a rebuilding visitor should _not_ be destructive.  That is to say that the originalScript that's passed in should not be modified in any way by any of our rebuilding visitors.  The changes should be evident in the rebuilt actualScript only.  

I will also add that this is definitely NOT the case with many of our rebuilding visitors.  They are certainly destructive (I know, because I've written a few that way, d'oh!).  I'm hoping that this is something that we can move away from.  If you're not feeling up to that for this PR, please do me a favor and make a separate issue so that we can track that as well.  

BTW, I loaded up your PR and added the validateLineage step against the originalScript...  This is definitely a destructive rebuilding visitor.  FYI.

Reply:
I will create a separate ticket to encapsulate this request to make it easier to track the effort for it across the relevant visitors.

Reply:
Thanks!

Reply:
I didn't see an issue yet, so I created one.